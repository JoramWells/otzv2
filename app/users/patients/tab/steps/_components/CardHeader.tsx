import React, { type ReactNode } from 'react'

interface CardHeaderProps {
  header: string
  rightContent?: ReactNode
}

const CardHeader = ({ header, rightContent }: CardHeaderProps) => {
  return (
    <div className="flex justify-between items-center w-full border-b bg-slate-50 border-slate-200  p-3  rounded-t-lg">
      <h3 className="text-lg  font-semibold">{header}</h3>
      {rightContent}
    </div>
  )
}

export default CardHeader
