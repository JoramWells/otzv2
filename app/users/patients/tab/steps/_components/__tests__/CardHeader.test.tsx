import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import CardHeader from '../CardHeader'

describe('CardHeader', () => {
  it('renders', () => {
    const headerText = 'Test Hear'
    const rightContent = 'Right Content'
    render(
      <CardHeader
        header={headerText}
        rightContent={<div>{rightContent}</div>}
      />
    )
    // const heading = screen.getByRole('heading', { level: 3 })

    const headerElement = screen.getByText(headerText)

    expect(headerElement).toBeInTheDocument()
    expect(headerElement).toHaveClass('text-lg font-semibold')

    const rightContentElement = screen.getByText(rightContent)
    expect(rightContentElement).toBeInTheDocument()
  })

  it('renders snapshot correctly', () => {
    const headerText = 'Test Hear'
    const rightContent = 'Right Content'
    const { container } = render(
          <CardHeader
            header={headerText}
            rightContent={<div>{rightContent}</div>}
          />
    )
    expect(container).toMatchSnapshot()
  })
})
