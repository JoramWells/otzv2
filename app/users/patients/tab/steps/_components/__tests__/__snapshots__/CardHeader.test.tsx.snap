// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader renders snapshot correctly 1`] = `
<div>
  <div
    class="flex justify-between items-center w-full border-b bg-slate-50 border-slate-200  p-3  rounded-t-lg"
  >
    <h3
      class="text-lg  font-semibold"
    >
      Test Hear
    </h3>
    <div>
      Right Content
    </div>
  </div>
</div>
`;
